src/main/java/com/qaprosoft/carina/core/foundation/cucumber/CucumberNameResolver.java

```diff
@@ -0,0 +1,92 @@
+package com.qaprosoft.carina.core.foundation.cucumber;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.testng.ITestResult;
+
+import com.qaprosoft.carina.core.foundation.commons.SpecialKeywords;
+import com.zebrunner.agent.testng.core.testname.TestNameResolver;
+
+import io.cucumber.testng.FeatureWrapper;
+import io.cucumber.testng.PickleWrapper;
+
+public class CucumberNameResolver implements TestNameResolver {
+
+    private final static String FEATURE_NAME_OPTIONAL = "Optional";
+
+    @Override
+    @SuppressWarnings({ "unlikely-arg-type" })
+    public String resolve(ITestResult result) {
+        String name = "";
+
+        if (result.getTestContext() == null) {
+            throw new RuntimeException("Unable to set Test name without testContext!");
+        }
+        @SuppressWarnings("unchecked")
+        Map<Object[], String> testnameMap = (Map<Object[], String>) result.getTestContext().getAttribute(SpecialKeywords.TEST_NAME_ARGS_MAP);
+
+        if (testnameMap != null) {
+            String testHash = String.valueOf(Arrays.hashCode(result.getParameters()));
+            if (testnameMap.containsKey(testHash)) {
+                name = testnameMap.get(testHash);
+            }
+        }
+
+        if (name.isEmpty()) {
+            name = result.getTestContext().getCurrentXmlTest().getName();
+        }
+
+        if (result.getTestContext().getCurrentXmlTest().getAllParameters().containsKey(SpecialKeywords.EXCEL_DS_CUSTOM_PROVIDER) ||
```

Comment:
do we need it for cucumber?

Reply:
I left it just in case. but don't think it will be used for cucumber. so I removed this logic at last